source/test/F0.Minesweeper.Logic.Tests/RandomToTest.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using F0.Minesweeper.Logic.Abstractions;
+
+namespace F0.Minesweeper.Logic.Tests
+{
+	internal class RandomToTest : IRandom
+	{
+		private int count;
+		private readonly int[] nextNumbers;
+		private readonly bool shouldRepeat;
+
+		public RandomToTest(int[] nextNumbers, bool shouldRepeat = false)
+		{
+			this.nextNumbers = nextNumbers;
+			this.shouldRepeat = shouldRepeat;
+		}
+
+		public int Next() => Next(int.MaxValue);
+
+		public int Next(int maxValue)
+		{
+			int nextToReturn;
+			if (shouldRepeat)
+			{
+				nextToReturn = nextNumbers[count % nextNumbers.Length];
+			}
+			else
+			{
+				nextToReturn = nextNumbers[count];
```

Comment:
Little improvement: rather than `IndexOutOfRange`ing, we could offer an explicit Exception (perhaps InvalidOperationException).
Or did you intend to cause an `IndexOutOfRangeException` here in case we don't repeat.